tryoutmanager/models.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python3.6
+
+import os
+import uuid
+import sqlalchemy as sa
+
+from datetime import datetime
+from sqlalchemy import Column, String, Boolean, DateTime, Table, \
+                       ForeignKey, Enum, Index
+from sqlalchemy.orm import relationship, sessionmaker
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.ext.declarative import declarative_base
+
+db = sa.create_engine(os.environ['DATABASE_URL'])
+metadata = sa.MetaData()
+
+Base = declarative_base(metadata)
+Session = sessionmaker(bind=db)
+
+
+# Association tables for many-to-many relationships
+markers_challenges = Table(
+        'markers_challenges', Base.metadata, Column(
+            'marker_email', String, ForeignKey('markers.email')),
+        Column('challenge_id', UUID, ForeignKey('challenges.id')))
+
+challenges_submissions = Table(
+        'challenges_submissions', Base.metadata, Column(
+                                   'challenge_id', UUID,
+                                   ForeignKey('challenges.id')),
+        Column('submission_id', UUID, ForeignKey('submissions.id')))
+
+
+# Tables corresponding to concrete models
+class AdminUser(Base):
+    __tablename__ = 'admin_users'
+
+    email = Column(String, nullable=False, unique=True, primary_key=True)
+    password = Column(String, nullable=False)
+    github_slug = Column(String, nullable=True)
+    public_keys = relationship('PublicKey', back_populates='user')
+    created_at = Column(DateTime(timezone=True), default=datetime.utcnow)
+
+
+class Marker(Base):
+    __tablename__ = 'markers'
+
+    email = Column(String, nullable=False, unique=True, primary_key=True)
+    password = Column(String, nullable=False)
+    name = Column(String, nullable=False)
+    challenges = relationship(
+            'Challenges', secondary=markers_challenges,
+            back_populates='markers')
+    github_slug = Column(String, nullable=True)
+    public_keys = relationship('PublicKey', back_populates='user')
+    created_at = Column(DateTime(timezone=True), default=datetime.utcnow)
+
+
+class User(Base):
+    __tablename__ = 'users'
+
+    email = Column(String, primary_key=True)
+    password = Column(String, nullable=False)
+    name = Column(String, nullable=False)
+    created_at = Column(DateTime(timezone=True), default=datetime.utcnow)
+    github_slug = Column(String, nullable=True)
+    public_keys = relationship('PublicKey', back_populates='user')
+    confirmed = Column(Boolean, nullable=False, default=False)
+    email_confirmation_slug = Column(String, unique=True)
+    email_confirmation_expiry = Column(DateTime(timezone=True))
+
+
+class PublicKey(Base):
```

Comment:
I might be missing something but doesn't this need to be associated with an owner?